pkg/handlers/ocm.go

```diff
@@ -113,13 +115,16 @@ func (b *ServiceLogBuilder) Build(firing bool, alert *template.Alert) (*ServiceL
 
 type OCMClient interface {
 	SendServiceLog(logEntry *slv1.LogEntry) error
+	SendLimitedSupport(clusterUUID string, lsReason *cmv1.LimitedSupportReason) error
+	RemoveLimitedSupport(clusterUUID string, lsReasonID string) error
+	GetLimitedSupportReasons(clusterUUID string) ([]*cmv1.LimitedSupportReason, error)
 }
 
 type ocmClientImpl struct {
 	ocmConnection *sdk.Connection
 }
 
-//go:generate mockgen -destination=mocks/service_logs.go -package=mocks github.com/openshift/ocm-agent/pkg/handlers OCMClient
+//go:generate mockgen -destination=mocks/ocm.go -package=mocks github.com/openshift/ocm-agent/pkg/handlers OCMClient
```

Comment:
I renamed this whole thing `ocm.go`, as it was already a wrapper for ocm and I didn't want to split it up in 2.

Reply:
@typeid - the way OA has been organised is kind of 'one handler per OCM API endpoint':
```
$ tree /home/hectorkemp/redhat/go/ocm-agent/pkg/handlers
/home/hectorkemp/redhat/go/ocm-agent/pkg/handlers
├── cluster.go
├── cluster_test.go
├── handlers_common.go
├── handlers_suite_test.go
├── handlers_test.go
├── helper.go
...
├── readyz_test.go
├── service_logs.go
├── service_logs_test.go
├── upgrade_policies.go
├── upgrade_policies_test.go
├── webhookreceiver.go
├── webhookreceiver_test.go
├── webhookrhobsreceiver.go
└── webhookrhobsreceiver_test.go
```
Where `cluster.go` corresponds to `/api/clusters_mgmt/v1/clusters`,
`upgrade_policies.go` corresponds to `/api/clusters_mgmt/v1/upgrade_policies`
`service_logs.go` to `/api/service_logs/v1/cluster_logs/{id}`.
To keep this ordering roughly consistent, could you pull your functions out into a `limited_support.go` handler?

Reply:
See this recently completed epic (https://issues.redhat.com/browse/SDE-2150) for further explanation on why we do/did this.

Reply:
Will address same comment below. Thanks!

Reply:
While I see what's trying to be done here, it doesn't make sense to me for the alertmanager webhooks. There's no direct 1:1 mapping from webhook to an OCM endpoint so ocm-agent can't be a "reverse proxy" as described in that epic. 

See e.g., I hope this shows the issue :
```
for _, service := range o.services {
			switch service {
			case config.ServiceLogService:
				o.logger.Info("Initialising alertmanager webhook handler in NON-fleet mode")
				webhookReceiverHandler := handlers.NewWebhookReceiverHandler(client, ocmclient)
				r.Path(consts.WebhookReceiverPath).Handler(webhookReceiverHandler)
				r.Use(metrics.PrometheusMiddleware)
			case config.ClustersService:
				o.logger.Info("Initialising UpgradePolicy handlers")
				upgradePolicyHandler := handlers.NewUpgradePoliciesHandler(sdkclient, internalID)
				// See https://github.com/gorilla/mux#examples
				r.HandleFunc("/upgrade_policies", upgradePolicyHandler.ServeUpgradePolicyList)
				r.HandleFunc("/upgrade_policies/{upgrade_policy_id}", upgradePolicyHandler.ServeUpgradePolicyGet)
				r.HandleFunc("/upgrade_policies/{upgrade_policy_id}/state", upgradePolicyHandler.ServeUpgradePolicyState)
				o.logger.Info("Initialising Cluster handlers")
				clusterHandler := handlers.NewClusterHandler(sdkclient, internalID)
				r.HandleFunc("/", clusterHandler.ServeClusterGet)
			}
```

Reply:
As discussed, moved `ocm.go` out into `pkg/ocm`.